spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/EmbeddedCassandra.scala

```diff
@@ -16,48 +16,146 @@ trait EmbeddedCassandra {
+          currentConfigTemplates = currentConfigTemplates.patch(i, Seq(configTemplates(i)), 1)
+        }
       }
     }
   }
 }
 
+
 object EmbeddedCassandra {
 
-  val DefaultHost = "127.0.0.1"
+  val HostProperty = "IT_TEST_CASSANDRA_HOSTS"
+  val NativePortProperty = "IT_TEST_CASSANDRA_NATIVE_PORTS"
+  val RpcPortProperty = "IT_TEST_CASSANDRA_RPC_PORTS"
+
+  validate
+
+  private def validate() = {
+    val hosts = sys.env.get(HostProperty)
+    if (hosts != None) {
+      val nativePorts = sys.env.get(NativePortProperty)
+      if (nativePorts != None) {
+        val rpcPorts = sys.env.get(RpcPortProperty)
+        if (rpcPorts != None) {
+          val hostSize = hosts.getOrElse("").split(",").size
+          val nativePortSize = nativePorts.getOrElse("").split(",").size
+          if (hostSize != nativePortSize) {
+            throw new RuntimeException("IT_TEST_CASSANDRA_HOSTS must have same size as IT_TEST_CASSANDRA_NATIVE_PORTS")
+          }
+          val rpcPortSize = rpcPorts.getOrElse("").split(",").size
+          if (hostSize != rpcPortSize) {
+            throw new RuntimeException("IT_TEST_CASSANDRA_HOSTS must have same size as IT_TEST_CASSANDRA_RPC_PORTS")
+          }
+        } else {
+          throw new RuntimeException("Missing IT_TEST_CASSANDRA_RPC_PORTS settings in system environment")
+        }
+      } else {
+        throw new RuntimeException("Missing IT_TEST_CASSANDRA_NATIVE_PORTS settings in system environment")
+      }
+    }
+  }
 
-  // TODO change to CASSANDRA_HOST
-  val HostProperty = "IT_TEST_CASSANDRA_HOST"
+  val hosts: IndexedSeq[InetAddress] = {
+    sys.env.get(HostProperty) match {
+      case Some(h) => h.split(",").map(host => InetAddress.getByName(host.trim)).toIndexedSeq
+      case None => IndexedSeq()
+    }
+  }
 
-  private[connector] var cassandra: Option[CassandraRunner] = None
+  val nativePorts: IndexedSeq[Int] = {
+    sys.env.get(HostProperty) match {
+      case Some(p) => p.split(",").map(port => port.trim.toInt).toIndexedSeq
+      case None => IndexedSeq()
+    }
+  }
 
-  private[connector] var currentConfigTemplate: String = null
+  val rpcPorts: IndexedSeq[Int] = {
+    sys.env.get(HostProperty) match {
+      case Some(p) => p.split(",").map(port => port.trim.toInt).toIndexedSeq
+      case None => IndexedSeq()
+    }
+  }
 
-  val cassandraHost = InetAddress.getByName(sys.env.getOrElse(HostProperty, DefaultHost))
+  private[connector] var cassandras: IndexedSeq[Option[CassandraRunner]] = IndexedSeq(None)
+
+  private[connector] var currentConfigTemplates: IndexedSeq[String] = IndexedSeq("")
+
+  def getProps(index: Integer): Map[String, String] = {
+    if (hosts.isEmpty || index < hosts.size) {
+      val host = if (hosts.isEmpty) "127.0.0.1" else hosts(index).getHostAddress
+      Map("seeds"               -> host,
+        "storage_port"          -> s"700$index",
```

Comment:
this is sooo minimalistic:) please make it `${7000 + index}`, same for the remaining settings